numpy/__init__.py

```diff
@@ -182,9 +182,36 @@
     oldnumeric = 'removed'
     numarray = 'removed'
 
-    # We don't actually use this ourselves anymore, but I'm not 100% sure that
-    # no-one else in the world is using it (though I hope not)
-    from .testing import Tester
+    if sys.version_info[:2] >= (3, 7):
+        # Importing Tester requires importing all of UnitTest which is not a
+        # cheap import Since it is mainly used in test suits, we lazy import it
+        # here to save on the order of 10 ms of import time for most users
+        #
+        # The previous way Tester was imported also had a side effect of adding
+        # the full `numpy.testing` namespace
+        def __getattr__(attr):
+            if attr == 'testing':
+                import numpy.testing as testing
+                return testing
+            elif attr == 'Tester':
+                from .testing import Tester
+                return Tester
+            else:
+                raise AttributeError(
+                    "module %s has no attribute $s".format(__name__, attr))
+
+
+        def __dir__():
+            # Why does 3.7 import these when you import *, but not add them
+            # in __all__???
+            not_added_to_all = ['bool', 'int', 'float', 'complex',
```

Comment:
Can you explain what you mean here? How are you concluding that this is happening

Reply:
I think the `test_numpy_namespace` test fails at finding `bool` and the re-exported built-in types without this.

I tried to add them to `__all__` using `extend`, but it didn't really work...???? that is why there are many question marks next to it.

Reply:
Apparently adding them to `__all__` works. I can make a selective change to python 3.7 or move it outside the if

Reply:
The easy test on the python console is:
```
In [1]: import numpy
In [2]: 'bool' in dir(numpy)
Out[2]: True
```
on master.